semcore/data-table/src/Head.tsx

```diff
@@ -48,10 +52,86 @@ class Head extends Component<AsProps> {
     }
   };
 
-  refColumn = (props: Column['props']) => (ref: HTMLElement) => {
-    setRef(props.ref, ref);
-    if (props.forwardRef) {
-      setRef(props.forwardRef, ref);
+  refSort = (active: boolean) => (ref: HTMLElement | null) => {
+    if (ref) {
+      this.sortWrapperRefs.set(ref, active);
+    }
+  };
+
+  refColumn = (column: Column) => (ref: HTMLElement | null) => {
+    setRef(column.props.ref, ref);
+    if (column.props.forwardRef) {
+      setRef(column.props.forwardRef, ref);
+    }
+
+    if (ref && ref.getAttribute('scope') === 'col') {
+      this.calculateMinWidth(ref, column);
+    }
+  };
+
+  calculateMinWidth = (node: HTMLElement, column: Column) => {
+    if (
+      !this.defaultMinWidths.has(node) &&
+      (column.props.wMin || column.props.wMax || column.props.w)
+    ) {
+      const computedStyle = window.getComputedStyle(node);
+
+      this.defaultMinWidths.set(node, cssToIntDefault(computedStyle.getPropertyValue('width')));
+    }
+
+    if (column.active) {
+      const clonedColumn = document.createElement('div');
+      const computedStyle = window.getComputedStyle(node);
+
+      node.childNodes.forEach((node) => {
+        if (!this.sortWrapperRefs.get(node)) {
+          clonedColumn.append(node.cloneNode(true));
+        }
+      });
+
+      const styles = [
+        'display',
+        'flex',
+        'margin',
+        'padding',
+        'background',
+        'font-style',
+        'font-width',
+        'font-size',
+        'font-weight',
+      ];
+
+      styles.forEach((key) => {
+        clonedColumn.style.setProperty(
+          key,
+          computedStyle.getPropertyValue(key),
+          computedStyle.getPropertyPriority(key),
+        );
+      });
+
+      clonedColumn.style.setProperty('visibility', 'hidden', 'important');
```

Comment:
Let's put it before `styles.forEach((key) => {`

Reply:
fixed